src/software/amazon/ionschema/internal/util/RangeInt.kt

```diff
@@ -0,0 +1,44 @@
+package software.amazon.ionschema.internal.util
+
+import software.amazon.ion.IonInt
+import software.amazon.ion.IonList
+import software.amazon.ion.IonSymbol
+import software.amazon.ionschema.InvalidSchemaException
+import java.math.BigDecimal
+
+/**
+ * Implementation of Range<Int>, which mostly delegates to RangeBigDecimal.
+ */
+internal class RangeInt (
+        private val ion: IonList,
+        private val delegate: RangeBigDecimal = RangeBigDecimal(ion)
+) : Range<Int> {
+
+    init {
+        if (!(ion[0] is IonInt
+                        || (ion[0] as? IonSymbol)?.stringValue().equals("min"))) {
```

Comment:
I see this being repeated enough I'd recommend an extension function:

```
fun IonValue?.isMinSymbol() = this as? IonSymbol?.stringValue() == "min"
```

Same for max.